Assets/Scripts/Layers/LayerTypes/GeoJSONPointLayer.cs

```diff
@@ -40,22 +43,63 @@ protected override void OnLayerActiveInHierarchyChanged(bool activeInHierarchy)
             pointRenderer3D.gameObject.SetActive(activeInHierarchy);
         }
 
-        public void AddAndVisualizeFeature(Feature feature, MultiPoint featureGeometry, CoordinateSystem originalCoordinateSystem)
+        public void AddAndVisualizeFeature<T>(Feature feature, CoordinateSystem originalCoordinateSystem)
+            where T : GeoJSONObject
         {
-            PointFeatures.Add(feature);
-            GeoJSONGeometryVisualizerUtility.VisualizeMultiPoint(featureGeometry, originalCoordinateSystem, PointRenderer3D);
+            // Skip if feature already exists (comparison is done using hashcode based on geometry)
+            if (SpawnedVisualisations.Any(f => f.feature.GetHashCode() == feature.GetHashCode()))
+                return;
+
+            var newFeatureVisualisation = new FeaturePointVisualisations() { feature = feature };
+        
+            if (feature.Geometry is MultiPoint multiPoint)
+            {
+                var newPointCollection = GeoJSONGeometryVisualizerUtility.VisualizeMultiPoint(multiPoint, originalCoordinateSystem, PointRenderer3D);
+                newFeatureVisualisation.pointCollection.Add(newPointCollection);
+            }
+            else if(feature.Geometry is Point point)
+            {
+                var newPointCollection = GeoJSONGeometryVisualizerUtility.VisualizePoint(point, originalCoordinateSystem, PointRenderer3D);
+                newFeatureVisualisation.pointCollection.Add(newPointCollection);
+            }
+
+            SpawnedVisualisations.Add(newFeatureVisualisation);
         }
 
-        public void AddAndVisualizeFeature(Feature feature, Point featureGeometry, CoordinateSystem originalCoordinateSystem)
+         /// <summary>
+        /// Checks the Bounds of the visualisations and checks them against the camera frustum
+        /// to remove visualisations that are out of view
+        /// </summary>
+        public void RemoveFeaturesOutOfView()
```

Comment:
ISn't this duplicate to the GeoJSONLineLayer? Similar to the RemoveFeature? If so: these methods do too much specific things to be duplicated. If there is a bug or improvement in one of these it is easy to forget to apply them to another

Reply:
I agree, I wanted to add a base class for them but they already extended from LayerNL3DBase so left it there.
Maybe we could move this to a utility or something.

Reply:
Will be new technical debt ticket; -,this can be refactored, probably making use of the new layer prefabs, instead of passing down the renderers from the adapters.